src/main/java/io/supertokens/webserver/api/passwordless/ConsumeCodeAPI.java

```diff
@@ -81,36 +82,46 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
+                    createRecipeUserIfNotExists);
 
             JsonObject result = new JsonObject();
             result.addProperty("status", "OK");
-            JsonObject userJson =
-                    getVersionFromRequest(req).greaterThanOrEqualTo(SemVer.v4_0) ? consumeCodeResponse.user.toJson() :
-                            consumeCodeResponse.user.toJsonWithoutAccountLinking();
 
-            if (getVersionFromRequest(req).lesserThan(SemVer.v3_0)) {
-                userJson.remove("tenantIds");
-            }
+            if (consumeCodeResponse.user != null) {
+                io.supertokens.useridmapping.UserIdMapping.populateExternalUserIdForUsers(this.getTenantIdentifierWithStorageFromRequest(req), new AuthRecipeUserInfo[]{consumeCodeResponse.user});
+
+                ActiveUsers.updateLastActive(this.getPublicTenantStorage(req), main, consumeCodeResponse.user.getSupertokensUserId());
+
+                JsonObject userJson = getVersionFromRequest(req).greaterThanOrEqualTo(SemVer.v4_0) ? consumeCodeResponse.user.toJson() :
+                        consumeCodeResponse.user.toJsonWithoutAccountLinking();
+
+                if (getVersionFromRequest(req).lesserThan(SemVer.v3_0)) {
+                    userJson.remove("tenantIds");
+                }
 
-            result.addProperty("createdNewUser", consumeCodeResponse.createdNewUser);
-            result.add("user", userJson);
-            if (getVersionFromRequest(req).greaterThanOrEqualTo(SemVer.v4_0)) {
-                for (LoginMethod loginMethod : consumeCodeResponse.user.loginMethods) {
-                    if (loginMethod.recipeId.equals(RECIPE_ID.PASSWORDLESS)
-                            && (consumeCodeResponse.email == null || Objects.equals(loginMethod.email, consumeCodeResponse.email))
-                            && (consumeCodeResponse.phoneNumber == null || Objects.equals(loginMethod.phoneNumber, consumeCodeResponse.phoneNumber))) {
-                        result.addProperty("recipeUserId", loginMethod.getSupertokensOrExternalUserId());
-                        break;
+                result.addProperty("createdNewUser", consumeCodeResponse.createdNewUser);
+                result.add("user", userJson);
+                if (getVersionFromRequest(req).greaterThanOrEqualTo(SemVer.v4_0)) {
+                    for (LoginMethod loginMethod : consumeCodeResponse.user.loginMethods) {
+                        if (loginMethod.recipeId.equals(RECIPE_ID.PASSWORDLESS)
+                                && (consumeCodeResponse.email == null || Objects.equals(loginMethod.email, consumeCodeResponse.email))
+                                && (consumeCodeResponse.phoneNumber == null || Objects.equals(loginMethod.phoneNumber, consumeCodeResponse.phoneNumber))) {
+                            result.addProperty("recipeUserId", loginMethod.getSupertokensOrExternalUserId());
+                            break;
+                        }
```

Comment:
- we also want to return the device
- remove the factorId stuff below + check that it's not there anywhere else too

Reply:
Done